sentry_streams/tests/adapters/arroyo/test_broadcaster.py

```diff
@@ -0,0 +1,201 @@
+            make_value_msg(
+                payload="test-payload",
+                route=Route(source="source", waypoints=["branch_1"]),
+                offset=0,
+            )
+        ),
+        call.submit(
+            make_value_msg(
+                payload="test-payload",
+                route=Route(source="source", waypoints=["branch_2"]),
+                offset=0,
+            )
+        ),
+    ]
+
+    broadcaster.submit(message)
+    next_step.assert_has_calls(expected_calls)
+
+
+def test_submit_filteredpayload() -> None:
+    next_step = mock.Mock(spec=ProcessingStrategy)
+    broadcaster = Broadcaster(
+        route=Route(source="source", waypoints=[]),
+        downstream_branches=["branch_1", "branch_2"],
+        next_step=next_step,
+    )
+
+    message = make_value_msg(
+        payload=FilteredPayload(), route=Route(source="source", waypoints=[]), offset=0
+    )
+
+    broadcaster.submit(message)
+    next_step.submit.assert_called_once_with(message)
+
+
+def test_submit_wrong_route() -> None:
+    next_step = mock.Mock(spec=ProcessingStrategy)
+    broadcaster = Broadcaster(
+        route=Route(source="source", waypoints=[]),
+        downstream_branches=["branch_1", "branch_2"],
+        next_step=next_step,
+    )
+
+    message = make_value_msg(
+        payload="wrong_route", route=Route(source="source", waypoints=["wrong"]), offset=0
+    )
+
+    broadcaster.submit(message)
+    next_step.submit.assert_called_once_with(message)
+
+
+def test_message_rejected() -> None:
+    next_step = mock.Mock(spec=ProcessingStrategy)
+    # raise MessageRejected on submit
+    next_step.submit.side_effect = MessageRejected()
+
+    broadcaster = Broadcaster(
+        route=Route(source="source", waypoints=[]),
+        downstream_branches=["branch_1", "branch_2"],
+        next_step=next_step,
+    )
+
+    message = make_value_msg(
+        payload="test-payload", route=Route(source="source", waypoints=[]), offset=0
+    )
+
+    expected_calls = [
+        # MessageRejected calls
+        call.submit(
+            make_value_msg(
+                payload="test-payload",
+                route=Route(source="source", waypoints=["branch_1"]),
+                offset=0,
+            )
+        ),
+        call.submit(
+            make_value_msg(
+                payload="test-payload",
+                route=Route(source="source", waypoints=["branch_2"]),
+                offset=0,
+            )
+        ),
+        # __flush_pending() calls
+        call.submit(
+            make_value_msg(
+                payload="test-payload",
+                route=Route(source="source", waypoints=["branch_1"]),
+                offset=0,
+            )
+        ),
+        call.submit(
+            make_value_msg(
+                payload="test-payload",
+                route=Route(source="source", waypoints=["branch_2"]),
+                offset=0,
+            )
+        ),
+    ]
+
+    broadcaster.submit(message)
```

Comment:
The contract is violated here. Your step should not swallow the MessageRejected exception otherwise Arroyo will not know it is in a backpressure state. It will only know that at the following message.